tutorials/20_Using_Haystack_with_REST_API.ipynb

```diff
@@ -0,0 +1,415 @@
+    "```bash\n",
+    "docker-compose up\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before continuing, test if everything is OK with the Haystack API by sending a cURL request to the `/initialized` endpoint. You can use the command line of your computer or tools like [Postman](https://www.postman.com/) to send cURL requests. If everything works fine, you will get `true` as a response.\n",
+    "\n",
+    "```bash\n",
+    "curl --request GET http://127.0.0.1:8000/initialized\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Index Files to Elasticsearch\n",
+    "\n",
+    "Right now, the Elasticsearch instance is empty. Haystack API provides a `/file-upload` endpoint to upload files to Elasticsearch using the indexing pipeline defined in the pipeline YAML. After indexing files to Elasticsearch, you will be able to perform document search."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Download example files.\n",
+    "\n",
+    "Download the [example files](https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/article_txt_countries_and_capitals.zip) you will be indexing to Elasticsearch. In the folder, there are text files about countries and capitals crawled from [Wikipedia](https://en.wikipedia.org/wiki/Category:Lists_of_countries_by_continent)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Index files to Elasticsearch.\n",
+    "\n",
+    "You can send cURL requests to the `/file-upload` endpoint to upload files to the Elasticsearch instance. Replace `<PATH_TO_FOLDER>` with the path to the example files on your computer and send the POST request. If the file is successfully uploaded, you will get `null` as a response.\n",
+    "\n",
+    "```bash\n",
+    "curl --request POST \\\n",
+    "     --url http://127.0.0.1:8000/file-upload \\\n",
+    "     --header 'accept: application/json' \\\n",
+    "     --header 'content-type: multipart/form-data' \\\n",
+    "     --form files=@<PATH_TO_FOLDER>0_Minsk.txt \\\n",
+    "     --form meta=null\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This method is not convenient for uploading multiple files to the Elasticsearch instance as replacing the file names in the request by hand is difficult. Instead, create a python file in the folder where you keep the example files, put the code below into the python file, and run the python script. This python code takes the name of every file in the folder and sends the POST request to the URL `http://127.0.0.1:8000/file-upload`. With the `print(response.text, \":\", file_name)` line, you will be able to see the names of all indexed files. Make sure you see the \"Completed\" text before continuing.\n",
+    "\n",
+    "```python\n",
+    "import os\n",
+    "import requests\n",
```

Comment:
this is going to be tricky: `requests` needs to be installed but I'm not sure we want to go through the rabbit hole of setting up a Python virtual env. I suggest we either use a "pure python" solution or a shell command, for example (not tested):
```
find ./files -name '*.txt' -exec curl -s -X POST -H 'Accept: application/json' -F files="@{}" http://127.0.0.1:8000/file-upload;
```